subsys/power/power.c

```diff
@@ -72,7 +72,11 @@ __weak void pm_power_state_exit_post_ops(struct pm_state_info info)
 	/*
 	 * This function is supposed to be overridden to do SoC or
 	 * architecture specific post ops after sleep state exits.
+	 *
+	 * The kernel expects that irqs are unlocked after this.
 	 */
+
+	irq_unlock(0);
```

Comment:
Since it has to be done anyway, can it be moved to the functions were this function is called, just after calling it?
Otherwise, I'd suggest to modify the function description to add that `irq_unlock` should be done in this function.

Reply:
This does not necessarily has to be done in this function, I can imagine a SoC that set `PRIMASK`, call `irq_unlock()` in `pm_power_state_set`  and unset `PRIMASK` before leave it and don't implement `pm_power_state_exit_post_ops`  or implement but doesn't need to restore interruption ... The fact is that currently, if one these functions are not implemented we have a problem because IRQs are not restored. I think these weak functions should be removed though.